app/src/common/shared/com/igalia/wolvic/ui/widgets/dialogs/RestartDialogWidget.java

```diff
@@ -53,6 +61,15 @@ public void updateUI() {
         setBody(getContext().getString(R.string.restart_dialog_text, getContext().getString(R.string.app_name)));
     }
 
+    @Override
+    public void onDismiss() {
+        if (noButtonsClicked && mCancelCallback != null) {
```

Comment:
What do you need the noButtonsClicked for?. If the dialog is dismissed we just execute the cancel callback if available and that's it. Am I missing anything?

Reply:
Currentlly `onDismiss()` is called when:
- Users dismiss explicitly (need to cancel)
- Restart Later is clicked (no need to cancel)
- Back is clicked (cancel has already been called before onDismiss())

That's why I needed `noButtonsClicked` here, i.e. for the last two cases cancel shouldn't be called in `onDismiss()`.

But now I realized the simpler solution may be:
- Replace `onDismiss()` with `hide(REMOVE_WIDGET)` in Restart Later click
- No need to call cancel before `onDismiss()` in Back click
- `onDismiss()` is called when users dismiss explicitly or Back is clicked

This way we don't need `noButtonsClicked` anymore. You agree this is a simpler way?

Reply:
Yes , please do so